pipeline/processor/postprocessor.py

```diff
@@ -227,7 +291,27 @@ def process(self, document: Document, filename=None) -> Document:
             duplicated_paths = [
                 path for path, count in path_counts.items() if count > 1
             ]
+
```

Comment:
One thing to not here is that the following path is duplicated, which isn't especially pythonic and could be refactored.

```for path in duplicated paths```

However, it was the quickest fix for the fact that the iterative merging process in merge_text_blocks deletes indices, so the indices calculated in get_paths_to_merge become stale throughout the loop. The solution was therefore to 
get the indices in the first loop, and perform merging in the second.